crates/ruma-state-res/src/event_auth.rs

```diff
@@ -156,7 +139,11 @@ pub fn auth_check<E: Event>(
 
     // Since v1, if type is m.room.create:
     if *incoming_event.event_type() == TimelineEventType::RoomCreate {
-        return check_room_create(incoming_event, room_version);
+        let room_create_event = RoomCreateEvent::new(incoming_event);
+
+        return check_room_create(room_create_event, room_version)
+            .map(|_| true)
+            .map_err(Error::custom);
```

Comment:
Personal preference, ignore if you want.
```suggestion
        return check_room_create(room_create_event, room_version)
            .map_or_else(Error::custom, |_| true);
```

Reply:
How about `.map_err(Error::custom)?` + `return true`, two statements? Feels like the simplest version.

Reply:
To be fair, this conversion is temporary until I fix the return type of `auth_check` to remove the boolean. I could add that commit on top here.

Reply:
Since this code is temporary, I'll ignore this and fix the return type in the next PR, if that's okay with you (same for the other comments).